src/parser/data/Literal.h

```diff
@@ -25,9 +25,8 @@ class Literal {
 
  public:
   template <typename T>
-  requires(ad_utility::Streamable<T> &&
-           !std::same_as<std::remove_cvref_t<T>, Literal>)
-  explicit Literal(T&& t)
+  requires(!std::same_as<std::remove_cvref_t<T>, Literal> &&
+           ad_utility::Streamable<T>) explicit Literal(T&& t)
```

Comment:
Swapped the order of these two, because otherwise this causes issues with the build because the compiler tries out different constructors.